github-rag/app.py

```
@@ -13,51 +15,117 @@
+            for key, value in st.session_state.file_cache.items():
+                try:
+                    del value
+                except Exception as e:
+                    logger.warning(f"Failed to cleanup cache entry {key}: {str(e)}")
+            st.session_state.file_cache.clear()
+        gc.collect()
+        logger.info("Session cleanup completed successfully")
     except Exception as e:
-        raise GitHubRAGError(f"Invalid repository URL: {str(e)}")
+        logger.error(f"Error during session cleanup: {str(e)}")
+        raise SessionError(f"Failed to cleanup session: {str(e)}")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)